code/game/machinery/computer/robot.dm

```diff
@@ -132,3 +163,29 @@
 					s.start()
 					D.visible_message("<span class='danger'>\the [D] self-destructs!</span>")
 					D.gib()
+		if("extract")
+			if(!extracting)
```

Comment:
early return

Reply:
also make it "busy" var and early return before the switch so the console is fully locked during it

Reply:
done